sdk/storage/inc/common/storage_credential.hpp

```diff
@@ -24,17 +30,19 @@ namespace Azure { namespace Storage {
       m_accountKey = std::move(accountKey);
     }
 
-    std::string AccountName;
+    const std::string AccountName;
 
   private:
     friend class SharedKeyPolicy;
-    std::string GetAccountKey()
+    friend struct Blobs::BlobSasBuilder;
```

Comment:
If Sas builders are decoupled from `SharedKeyCredential`, we don't need this anymore. Please consider it.

Reply:
i'd rather not do that. see comments https://github.com/Azure/azure-sdk-for-cpp/pull/353#discussion_r460931362.